cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-media.php

```diff
@@ -1235,65 +1231,110 @@ public function balance_crop( $size, $shift_size ) {
-				$sync_key .= $file_info['extension'];
-			}
-			// Try to find the Attachment ID in context meta data.
-			$attachment_id = $this->get_id_from_sync_key( $sync_key );
-
-			if ( empty( $attachment_id ) ) {
-				// ensure the url is correct.
-				$asset['secure_url'] = $url;
-				$attachment_id       = $this->create_attachment( $asset, $public_id );
-				$return              = array(
-					'fetch'         => rest_url( REST_API::BASE . '/asset' ),
-					'uploading'     => true,
-					'src'           => $src,
-					'url'           => $url,
-					'filename'      => basename( $src ),
-					'attachment_id' => $attachment_id,
-				);
+			$asset = $this->get_asset_payload();
+			// Set a base array for pulling an asset if needed.
+			$base_return = array(
+				'fetch'         => rest_url( REST_API::BASE . '/asset' ),
+				'uploading'     => true,
+				'src'           => $asset['src'],
+				'url'           => $asset['url'],
+				'filename'      => basename( $asset['src'] ),
+				'attachment_id' => $asset['attachment_id'],
+			);
+			if ( empty( $asset['attachment_id'] ) ) {
+				$return                  = $base_return;
+				$asset['attachment_id']  = $this->create_attachment( $asset, $asset['public_id'] );
+				$return['attachment_id'] = $asset['attachment_id'];
 			} else {
-				$return              = wp_prepare_attachment_for_js( $attachment_id );
-				$return['public_id'] = $public_id;
+				// Capture the ALT Text.
+				if ( ! empty( $asset['meta']['alt'] ) ) {
+					$alt_text = wp_strip_all_tags( $asset['meta']['alt'] );
+					update_post_meta( $asset['attachment_id'], '_wp_attachment_image_alt', $alt_text );
+				}
+				// Compare Version.
+				$current_version = (int) $this->get_post_meta( $asset['attachment_id'], Sync::META_KEYS['version'], true );
+				if ( $current_version !== $asset['version'] ) {
+					// Difference version, remove files, and downsync new files related to this asset.
+					// If this is a different version, we should try find attachments with the base sync key and update the source.
+					$ids    = $this->get_id_from_sync_key( 'base_' . $asset['public_id'], true );
+					$resync = array();
+					foreach ( $ids as $id ) {
+						$meta = wp_get_attachment_metadata( $id );
+						wp_delete_attachment_files( $id, $meta, array(), get_attached_file( $id ) );
+						$this->update_post_meta( $id, Sync::META_KEYS['version'], $asset['version'] );
+						// Set signature for storage, since this will be more effective at downloading or not.
+						$this->sync->set_signature_item( $id, 'storage', '' );
+						if ( $id !== $asset['attachment_id'] ) {
+							$resync[] = wp_prepare_attachment_for_js( $id );
+						}
+					}
+					// Use the primary ID as the main return, and add the resynced assets to that.
+					$return           = wp_prepare_attachment_for_js( $asset['attachment_id'] );
+					$return['resync'] = $resync;
+				} else {
+					$return              = wp_prepare_attachment_for_js( $asset['attachment_id'] );
```

Comment:
Since this line is being duplicated, we can add it on top, outside of the of the `if`

Reply:
No we can't. I don't think to do that, but meets data is being changed, so it has to be as late as possible.